src/nbl/video/CVulkanCommandBuffer.cpp

```diff
@@ -48,25 +48,41 @@ void fill(vk_barrier_t& out, const ResourceBarrier& in, uint32_t selfQueueFamily
     // https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-VkBufferMemoryBarrier2-buffer-04088
     if (concurrentSharing)
         selfQueueFamilyIndex = IQueue::FamilyIgnored;
+
+    auto mapQFIdx = [](uint32_t idx)
+    {
+        switch (idx)
+        {
+        case IQueue::FamilyExternal:
+        case IQueue::FamilyIgnored:
+        case IQueue::FamilyForeign:
+            idx |= 1u << 31;
```

Comment:
this is kinda flaky, cause it relies on my encoding I have that I only use 1 bit for something else, I'd rather do a real switch-case-return (the VK enum)